### PR TITLE
Enabling python unit tests

### PR DESCRIPTION
**Pull request by cbernet**

### CODE REVIEW

**Comment by cbernet** [comment]
retest this please

**Comment by cbernet** [comment]
retest this please

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -39,12 +39,14 @@ endforeach( sourcefile ${executables} )
...
 add_test(NAME write COMMAND write)
 set_property(TEST write PROPERTY ENVIRONMENT LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/examples:$ENV{LD_LIBRARY_PATH})
 
 add_test(NAME read COMMAND read DEPENDS write)
 set_property(TEST read PROPERTY ENVIRONMENT LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/examples:$ENV{LD_LIBRARY_PATH})
 
-add_test(NAME read_python COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/read.py)
-set_property(TEST read_python PROPERTY ENVIRONMENT PYTHONPATH=${CMAKE_INSTALL_PREFIX}/python:${CMAKE_INSTALL_PREFIX}/examples:$ENV{PYTHONPATH}
-                                                   LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/examples:$ENV{LD_LIBRARY_PATH})
+add_test( NAME pyunittest COMMAND python -m unittest discover -s ${CMAKE_INSTALL_PREFIX}/python)
```
Comment (jlingema) [inline examples/CMakeLists.txt]: Looking at the unit test that is currently there, I think this should be dependent on write as well (same as read)? Although, you try again to write if the file is not there...

Comment (cbernet) [inline examples/CMakeLists.txt]: Hi Joschka, 
Actually that's something I wanted to discuss with you. At least on my mac, DEPENDS is not a valid parameter or add_test. I tried to put read before write and it fails. In pyunittest, as a consequence, DEPENDS is considered as an argument to the python executable, and that's how I noticed the issue. 
Would you like to merge to enable the test again? I'll contact you on skype to discuss the DEPENDS issue.

**Pull request merged**